### PR TITLE
[FEATURE] Auto-add system fields showitem for tt_content

### PR DESCRIPTION
**Pull request by linawolf**
References: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/1013
Releases: main

### CODE REVIEW

File: Documentation/Types/TtContent.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+..  figure:: /Images/ManualScreenshots/tt_content_extended.png
+
+Additional fields that are subsequently added to the end of the table using
+:php:`ExtensionManagementUtility::addToAllTcaTypes()` will appear in the tab
+:guilabel:`Extended`.
+
+..  _types-content-migration:
+
+Migration: Remove system fields from showitems on dropping TYPO3 12.4 support
```
Comment (brotkrueml) [inline Documentation/Types/TtContent.rst:94]:
```suggestion
Migration: Remove system fields from showitems on dropping TYPO3 v12.4 support
```

File: Documentation/Types/TtContent.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+=============================================================================
```
Comment (brotkrueml) [inline Documentation/Types/TtContent.rst:95]:
```suggestion
==============================================================================
```

File: Documentation/Types/TtContent.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+It is sufficient to apply changes to the :confval:`types-showitem` section
+of content types once dropping TYPO3 12.4 support in extensions.
```
Comment (brotkrueml) [inline Documentation/Types/TtContent.rst:98]:
```suggestion
of content types once dropping TYPO3 v12.4 support in extensions.
```

**Review by brotkrueml — DISMISSED** [review]

**Comment by linawolf** [comment]
Feel free to commit such minor changes yourself next time ;)

File: Documentation/Types/TtContent.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+Migration: Remove system fields from showitems on dropping TYPO3 v12.4 support
+==============================================================================
+
+It is sufficient to apply changes to the :confval:`types-showitem` section
+of content types once dropping TYPO3 v12.4 support in extensions.
+
+In site packages or extensions only supporting TYPO3 v13.3 or above you can
+migrate the :confval:`types-showitem` right away:
+
+..  literalinclude:: _CodeSnippets/tt_content_migration.diff
```
Comment (brotkrueml) [inline Documentation/Types/TtContent.rst:103]: Please add a caption for easy reception where to put this into.

And: is `:language: php` not missing?

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/Types/TtContent.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+Examples for the `showitems` TCA section in content elements
+============================================================
+
+..  _types-content-examples-basic:
+
+Basic custom content element with header and bodytext
+-----------------------------------------------------
+
+..  literalinclude:: _CodeSnippets/_basic_content_element.php
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:70]: language php is autodetected from the file eding

File: Documentation/Types/TtContent.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+The following tabs are shown, the header palette and bodytext field are shown
+in palette general:
+
+..  figure:: /Images/ManualScreenshots/tt_content_basic.png
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:75]: language php is autodetected from the file eding

File: Documentation/Types/TtContent.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+=================================================================
+Automatically added system fields to content types (`tt_content`)
+=================================================================
+
+..  versionchanged:: 13.3
+    Creating content elements has been simplified by removing the need to
+    define the system fields for each element again and again. This shrinks
+    down a content element's :confval:`types-showitem` to just the element
+    specific fields. See also :ref:`Migration <types-content-migration>`.
+    Added with :ref:`changelog:feature-104814-1725444916`.
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:16]: To me it is an information that is helpful in the versionchanged info (which will be removed after a version or 2). When I have the changelog entry I can also find the change on Gerrit, see which files where changed etc.

File: Documentation/Types/TtContent.rst [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+-------------------------------------------
+
+..  literalinclude:: _CodeSnippets/_extended_content_element.php
+
+The following tabs are shown:
+
+..  figure:: /Images/ManualScreenshots/tt_content_extended.png
+
+Additional fields that are subsequently added to the end of the table using
+:php:`ExtensionManagementUtility::addToAllTcaTypes()` will appear in the tab
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:89]: php-short only works with pure FQN. For anything else it does the same like :php

File: Documentation/Types/TtContent.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/Types/TtContent.rst:111]: Caption is added, language is auto-detected

File: Documentation/Types/TtContent.rst [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+It is sufficient to apply changes to the :confval:`types-showitem` section
+of content types once dropping TYPO3 v12.4 support in extensions.
+
+In site packages or extensions only supporting TYPO3 v13.3 or above you can
+migrate the :confval:`types-showitem` right away:
+
+..  literalinclude:: _CodeSnippets/tt_content_migration.diff
+
+So the tabs :guilabel:`General`, :guilabel:`Language`, :guilabel:`Access`
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:114]: Good idea

File: Documentation/Types/TtContent.rst [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+..  _types-content-examples-extended:
+
+Extended content element with custom fields
+-------------------------------------------
+
+..  literalinclude:: _CodeSnippets/_extended_content_element.php
+
+The following tabs are shown:
+
+..  figure:: /Images/ManualScreenshots/tt_content_extended.png
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:91]: done

File: Documentation/Types/TtContent.rst [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+in palette general:
+
+..  figure:: /Images/ManualScreenshots/tt_content_basic.png
+
+..  _types-content-examples-extended:
+
+Extended content element with custom fields
+-------------------------------------------
+
+..  literalinclude:: _CodeSnippets/_extended_content_element.php
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:86]: auto-detected

File: Documentation/Types/TtContent.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst:76]: caption was added

File: Documentation/Types/TtContent.rst [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+:confval:`types-showitem`. In this case, the :guilabel:`General` tab will be omitted.
+
+All those system fields, which are added based on the :php:`ctrl` section are
+also automatically removed from any custom palette and from the customized
+type's :confval:`types-showitem` definition.
+
+If the content element defines the :guilabel:`Extended` tab, it will be
+inserted at the end, including all fields added to the type via API methods,
+without specifying a position, via
+:php:`ExtensionManagementUtility::addToAllTcaTypes()`. See
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst]: Good point

File: Documentation/Types/Index.rst [hunk 13]
```diff
@@ -20,6 +20,18 @@ programming.
 ..  contents:: Table of Contents
     :depth: 1
 
+..  toctree:: Subpages
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/Index.rst]: yeah, I always mix that up

File: Documentation/Types/TtContent.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/Types/TtContent.rst]: Yes, it is not all about? ExtensionManagementUtility was mentioned before, so using the shortcut is okay. But with FQCN and php-short more information is displayed in the tooltip. or am I wrong? You know, I like consistency. And in another PR this was done that way (doesn't remember currently which one).

[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst]: php-short can anlyze FQN class names. It cannot however analyze a class name in combintation with something else. and here we have a class name, followed by a method name. We can try to enhance the php-short role but for now it only works for class (interface etc) names.

[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst]: See https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/Reference/ReStructuredText/Links/Api.html#links-api-short for reference

[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Types/TtContent.rst]: I therefore used the php text role

**Comment by linawolf** [comment]
@brotkrueml I addressed all your remarks, can we merge this in your eyes?

**Pull request merged**